### PR TITLE
HPCC-19358 Add new attributes to ESP configuration

### PR DESCRIPTION
**Pull request by kenrowland**
Add maxPersistentIdleTime and maxPersistentRequests to ESP configuration

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by kenrowland** [comment]
@mayx Please review. 

Also comment if we need a documentation change for this or not. Pay close attention to the tooltips to make sure they are correct.

**Comment by kenrowland** [comment]
@JamesDeFabia Please review tooltips

File: initfiles/componentfiles/configxml/esp.xsd.in [hunk 1]
```diff
@@ -937,6 +936,20 @@
                     </xs:appinfo>
                 </xs:annotation>
             </xs:attribute>
+            <xs:attribute name="maxPersistentIdleTime" type="xs:integer" use="optional" default="60">
+                <xs:annotation>
+                    <xs:appinfo>
+                        <tooltip>The maximum idle time (-1 for unlimited, 0 to disable).</tooltip>
```
Comment (JamesDeFabia) [inline initfiles/componentfiles/configxml/esp.xsd.in]: Time (milliseconds) to wait for additional requests on a persistent http connection. (-1 for unlimited, 0 to disable).

File: initfiles/componentfiles/configxml/esp.xsd.in [hunk 2]
```diff
@@ -937,6 +936,20 @@
...
+                    </xs:appinfo>
+                </xs:annotation>
+            </xs:attribute>
+            <xs:attribute name="maxPersistentRequests" type="xs:integer" use="optional" default="100">
+                <xs:annotation>
+                    <xs:appinfo>
+                        <tooltip>The maximum number of persistent requests (-1 for unlimited, 0 to disable).</tooltip>
```
Comment (JamesDeFabia) [inline initfiles/componentfiles/configxml/esp.xsd.in]: Maximum number of query requests per persistent http connection. (-1 for unlimited, 0 to disable).

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Comments inline

File: initfiles/componentfiles/configxml/esp.xsd.in (hunk 1 above)
Comment (kenrowland) [reply initfiles/componentfiles/configxml/esp.xsd.in]: Would (ms) suffice? @JamesDeFabia

Comment (mayx) [reply initfiles/componentfiles/configxml/esp.xsd.in]: @JamesDeFabia @kenrowland Here its number of seconds, not milli-seconds

Comment (JamesDeFabia) [reply initfiles/componentfiles/configxml/esp.xsd.in]: @mayx In Roxie.xsd.in, the attribute uses milliseconds. Should we be consistent?

Comment (kenrowland) [reply initfiles/componentfiles/configxml/esp.xsd.in]: How about this: The maximum idle time in seconds (-1 for unlimited, 0 to disable).

I don't see a maxPersistentIdleTime in roxie, perhaps I am missing the equivalent?

Comment (JamesDeFabia) [reply initfiles/componentfiles/configxml/esp.xsd.in]: maxHttpKeepAliveWait in roxie.xsd.in 
https://github.com/hpcc-systems/HPCC-Platform/blob/master/initfiles/componentfiles/configxml/roxie.xsd.in

Comment (mayx) [reply initfiles/componentfiles/configxml/esp.xsd.in]: @JamesDeFabia you can raise another issue if you think milli-seconds makes more sense. But right now the code in ESP (already merged in) is in seconds and the configuration should conform to that. (by the way this part of ESP code got in before Roxie :) @kenrowland please keep it as seconds to conform to ESP code.

**Comment by kenrowland** [comment]
@JamesDeFabia @mayx Please mark as approved

**Review by mayx — APPROVED** [review]
@kenrowland looks good.

**Review by JamesDeFabia — APPROVED** [review]

**Comment by kenrowland** [comment]
@richardkchapman Please merge

**Pull request merged**